### PR TITLE
Add string utils

### PR DESCRIPTION
**Pull request by AnubhavUjjawal**
Added a [StringUtils package](https://github.com/AnubhavUjjawal/Chapel-StringUtils) which Closes https://github.com/chapel-lang/chapel/issues/11433

### CODE REVIEW

**Comment by ben-albrecht** [comment]
@AnubhavUjjawal - Thanks for submitting this package. It is currently being reviewed.

**Comment by ben-albrecht** [comment]
Hi @AnubhavUjjawal, here are a few high-level comments on your package / interface:

- You may want to add some tests for some tricky unicode strings.
  - e.g. how should `strToArr("«ταБЬℓσ»: 1<2 & 4+1>3, foo")` work?
  - You may want an additional arg to handle how to treat characters, see [string docs](https://chapel-lang.org/docs/master/builtins/String.html#lengths-and-offsets-in-unicode-strings) for more info.
- I'm not sure you need an `in` argument intent for `isFullString(obj)`, since you're not modifying `obj`.
- Some module-level variables and procedures are likely not intended for the public interface. You can mark them with `private` so that they cannot be accessed outside of the module and will not show up in documentation.
- Stylistic:
  - Chapel generally uses `camelCase` for variables and procedure names, including module-level variables, rather than `ALLCAPS` and/or `under_scores`.
  - Chapel generally uses 2-space indentation

You're welcome to make some of these changes or leave it as is. Let me know when you feel happy with your version, and I'll merge your package into the registry. Note that any updates will require you to update the `v0.1.0` version tag.

**Comment by AnubhavUjjawal** [comment]
@ben-albrecht  Thanks for the review. I will make the updates as soon as possible, and will let you know. I agree with your points totally.

**Comment by AnubhavUjjawal** [comment]
@ben-albrecht 

> Hi @AnubhavUjjawal, here are a few high-level comments on your package / interface:
>    
>   * e.g. how should `strToArr("«ταБЬℓσ»: 1<2 & 4+1>3, foo")` work?

strToArr() returns a array with each character of string separately in array. I have made that available in the documentation. I am confused what you asked for?

**Comment by AnubhavUjjawal** [comment]
> * You may want an additional arg to handle how to treat characters, see [string docs](https://chapel-lang.org/docs/master/builtins/String.html#lengths-and-offsets-in-unicode-strings) for more info.

I would like to leave it on default for now. Maybe later, if asked for or really needed, I would update it?

**Comment by ben-albrecht** [comment]
> strToArr() returns a array with each character of string separately in array. I have made that available in the documentation. I am confused what you asked for?

Because string size and iterators default to bytes, unicode strings like the one above will give some unexpected results, e.g.

```chpl
var s = "«ταБЬℓσ»";

writeln('size: ', s.size); // Prints 17 (bytes) instead of 8 (code points)
```

> I would like to leave it on default for now. Maybe later, if asked for or really needed, I would update it?

That works. This was all optional feedback anyway. Thanks!

File: Bricks/StringUtils/0.1.0.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (ben-albrecht) [inline Bricks/StringUtils/0.1.0.toml]: Note: You need to update this file name to `0.1.1.toml`

**Review by ben-albrecht — CHANGES_REQUESTED** [review]

File: Bricks/StringUtils/0.1.0.toml (hunk 1 above)
Comment (AnubhavUjjawal) [reply Bricks/StringUtils/0.1.0.toml]: @ben-albrecht  carelessness from my side, sorry for that.

Comment (AnubhavUjjawal) [reply Bricks/StringUtils/0.1.0.toml]: I think it's fine now. Please let me know.

**Review by ben-albrecht — APPROVED** [review]
Thanks!
